peripherals/ucpd/ucpd_v1.yaml

```
@@ -0,0 +1,201 @@
+# UCPD peripheral
+
+CFGR1:
+  UCPDEN:
+    Disabled: [0, "UCPD peripheral disabled"]
+    Enabled: [1, "UCPD peripheral enabled"]
+  RXDMAEN:
+    Disabled: [0, "DMA mode for reception disabled"]
+    Enabled: [1, "DMA mode for reception enabled"]
+  TXDMAEN:
+    Disabled: [0, "DMA mode for transmission disabled"]
+    Enabled: [1, "DMA mode for transmission enabled"]
+  _split: [RXORDSETEN]
+  RXORDSETEN0:
+    Disabled: [0, "SOP detect disabled"]
+    Enabled: [1, "SOP detect enabled"]
+  RXORDSETEN1:
+    Disabled: [0, "SOP' detect disabled"]
+    Enabled: [1, "SOP' detect enabled"]
+  RXORDSETEN2:
+    Disabled: [0, "SOP'' detect disabled"]
+    Enabled: [1, "SOP'' detect enabled"]
+  RXORDSETEN3:
+    Disabled: [0, "Hard Reset detect enabled"]
+    Enabled: [1, "Hard Reset detect enabled"]
+  RXORDSETEN4:
+    Disabled: [0, "Cable Detect reset enabled"]
+    Enabled: [1, "Cable Detect reset enabled"]
+  RXORDSETEN5:
+    Disabled: [0, "SOP'_Debug disabled"]
+    Enabled: [1, "SOP'_Debug enabled"]
+  RXORDSETEN6:
+    Disabled: [0, "SOP''_Debug disabled"]
+    Enabled: [1, "SOP''_Debug enabled"]
+  RXORDSETEN7:
+    Disabled: [0, "SOP extension #1 disabled"]
+    Enabled: [1, "SOP extension #1 enabled"]
+  RXORDSETEN8:
+    Disabled: [0, "SOP extension #2 disabled"]
+    Enabled: [1, "SOP extension #2 enabled"]
```

Only one note.
I would prefer description to be in field description.
With 1 common enum.
```suggestion
  # Move this part to another file in common_patches
  _delete: RXORDSETEN
  _add:
    RXORDSETEN0:
      description: SOP detection
      bitOffset: 20
      bitWidth: 1
    RXORDSETEN1:
      description: SOP' detection
      bitOffset: 21
      bitWidth: 1
    RXORDSETEN2:
      description: SOP'' detection
      bitOffset: 22
      bitWidth: 1
    RXORDSETEN3:
      description: Hard Reset detection
      bitOffset: 23
      bitWidth: 1
    RXORDSETEN4:
      description: Cable Detect reset
      bitOffset: 24
      bitWidth: 1
    RXORDSETEN5:
      description: SOP'_Debug
      bitOffset: 25
      bitWidth: 1
    RXORDSETEN6:
      description: SOP'' Debug
      bitOffset: 26
      bitWidth: 1
    RXORDSETEN7:
      description: "SOP extension #1" 
      bitOffset: 27
      bitWidth: 1
    RXORDSETEN8:
      description: "SOP extension #1"
      bitOffset: 28
      bitWidth: 1
  
  # Leave this here
  RXORDSETEN?:
    Disabled: [0, Flag disabled]
    Enabled: [1, Flag enabled]
```

---

From where should I include that RXORDSETEN patch? In every `stm32xxx.yaml` that has UCPD or can I just put it once in `ucpd_v1.yaml`?

---

Ideally `peripheral` folder should contain only addition of enumeratedValues and writeConstains.
This is why I've asked to move it.
Although I does not insist.